lib/modules/manager/gradle-wrapper/utils.ts

```diff
@@ -41,13 +42,24 @@ export async function prepareGradleCommand(
  * Find compatible java version for gradle.
  * see https://docs.gradle.org/current/userguide/compatibility.html
  * @param gradleVersion current gradle version
+ * @param gradlewFile path to gradle wrapper
  * @returns A Java semver range
  */
-export function getJavaConstraint(
+export async function getJavaConstraint(
   gradleVersion: string | null | undefined,
-): string {
+  gradlewFile: string,
+): Promise<string> {
   const major = gradleVersion ? gradleVersioning.getMajor(gradleVersion) : null;
   const minor = gradleVersion ? gradleVersioning.getMinor(gradleVersion) : null;
+
+  // https://docs.gradle.org/8.8/release-notes.html#daemon-toolchains
+  if (major && major >= 8 && minor && minor >= 8) {
```

Comment:
```suggestion
  if (major && (major > 8 || (major === 8 && minor && minor >= 8))) {
```

This suggested change assumes you want it to work immediately if/when v9 comes out

Reply:
Thank you for spotting this!